torch/ao/quantization/pt2e/utils.py

```diff
@@ -11,12 +11,83 @@
 from typing import Any, Callable, Dict, Optional, Tuple, List, Union
 from torch.utils._pytree import LeafSpec
 
+from torch.ao.quantization.quantizer import QuantizationAnnotation
+from torch.ao.quantization.quantizer.utils import _is_sym_size_node
+
 __all__ = [
     "fold_bn_weights_into_conv_node",
     "get_aten_graph_module",
     "remove_tensor_overload_for_qdq_ops",
+    "_filter_sym_size_users",
+    "_find_q_dq_node_for_user",
+    "_is_valid_annotation",
+]
+
+_QUANTIZE_OPS = [
+    torch.ops.quantized_decomposed.quantize_per_tensor.default,
+    torch.ops.quantized_decomposed.quantize_per_tensor.tensor,
+    torch.ops.quantized_decomposed.quantize_per_channel.default,
 ]
 
+
+_DEQUANTIZE_OPS = [
+    torch.ops.quantized_decomposed.dequantize_per_tensor.default,
+    torch.ops.quantized_decomposed.dequantize_per_tensor.tensor,
+    torch.ops.quantized_decomposed.dequantize_per_channel.default,
+]
+
+
+def _is_connected(next_node: torch.fx.Node, target: torch.fx.Node) -> bool:
+    if target.op == "output":
+        return False
+    if next_node == target:
+        return True
+    for n in next_node.users.keys():
+        if _is_connected(n, target):
+            return True
+    return False
+
+
+def _find_q_dq_node_for_user(
```

Comment:
I feel this will be hard to debug and maintain, what is the problem with the existing duplicate dq implementation: https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/pt2e/qat_utils.py#L580?

Reply:
oh I looked again, is this for test only? can you move this to test files?

Reply:
No this is also used later in the stack for metdata porting.